src/frontend/Sider/Behandling/Vilkårvurdering/EndreVilkår/EndreVilkårPerioder.tsx

```diff
@@ -0,0 +1,180 @@
+import React from 'react';
+
+import { useFlag } from '@unleash/proxy-client-react';
+import styled from 'styled-components';
+
+import { HStack, Switch } from '@navikt/ds-react';
+
+import DateInputMedLeservisning from '../../../../komponenter/Skjema/DateInputMedLeservisning';
+import MonthInput from '../../../../komponenter/Skjema/MonthInput';
+import { MånedÅrVelger } from '../../../../komponenter/Skjema/MånedÅrVelger/MånedÅrVelger';
+import TextField from '../../../../komponenter/Skjema/TextField';
+import { FeilmeldingMaksBredde } from '../../../../komponenter/Visningskomponenter/FeilmeldingFastBredde';
+import { tilFørsteDagenIMåneden, tilSisteDagenIMåneden } from '../../../../utils/dato';
+import { harTallverdi, tilHeltall } from '../../../../utils/tall';
+import { Toggle } from '../../../../utils/toggles';
+import { fjernSpaces } from '../../../../utils/utils';
+import { Feilmeldinger } from '../validering';
+
+const StyledSwitch = styled(Switch)`
+    align-self: end;
+`;
+
+interface EndreVilkårPerioderProps {
+    fom: string | undefined;
+    tom: string | undefined;
+    feilmeldinger: Feilmeldinger;
+    erMidlertidigOvernatting: boolean;
+    utgift: number | undefined;
+    alleFelterKanRedigeres: boolean;
+    erNullvedtak: undefined | boolean;
+    settFom: (fom: string | undefined) => void;
+    settTom: (tom: string | undefined) => void;
+    settFeilmeldinger: (
+        value: ((prevState: Feilmeldinger) => Feilmeldinger) | Feilmeldinger
+    ) => void;
+    settDetFinnesUlagredeEndringer: (value: ((prevState: boolean) => boolean) | boolean) => void;
+    settUtgift: (
+        value: ((prevState: number | undefined) => number | undefined) | number | undefined
+    ) => void;
+    settErNullvedtak: (
+        value: ((prevState: boolean | undefined) => boolean | undefined) | boolean | undefined
+    ) => void;
+}
+
+const EndreVilkårPerioder = ({
+    fom,
+    settFom,
+    tom,
+    settTom,
+    erMidlertidigOvernatting,
+    settFeilmeldinger,
+    feilmeldinger,
+    alleFelterKanRedigeres,
+    settDetFinnesUlagredeEndringer,
+    utgift,
+    settUtgift,
+    erNullvedtak,
+    settErNullvedtak,
+}: EndreVilkårPerioderProps) => {
+    const skalBrukeMånedÅrVelger = useFlag(Toggle.SKAL_BRUKE_MANED_AR_VELGER);
+
+    const oppdaterErNullvedtak = (erNullvedtak: boolean) => {
+        settUtgift(undefined);
+        settErNullvedtak(erNullvedtak);
+    };
+
+    return (
+        <HStack gap="4" align="start">
+            <FeilmeldingMaksBredde $maxWidth={152}>
+                {erMidlertidigOvernatting ? (
```

Comment:
Kan man bryte ut datovelgeren? 
Og sende med et flagg som sier hvilken type som er ønskelig? 
Då trenger vi ikke en `erMidlertidigOvernatting` til en felleskomponent som er spesifik for boutgifter